src/components/Maps/MapPage.tsx

```diff
@@ -288,51 +533,192 @@ const MapPage = ({
+            longitude: lng,
+            address,
+            pinType: PinType.EVENT,
+          },
+        })
+      );
+    } else if (mode === MapModes.EditEvent) {
+      const eventPinId = pins.filter(
+        (pin) => pin.eventId.id === eventId && pin.pinType === PinType.EVENT
+      )[0].id;
+      editPin({
+        variables: {
+          id: eventPinId,
+          label: name,
+          eventId,
+          latitude: lat,
+          longitude: lng,
+          address,
+        },
+      });
+      editEvent({
+        variables: {
+          id: eventId,
+          name,
+          eventDate,
+          createdBy: 1,
+          isActive: true,
+        },
+      });
+    }
+  };
+
+  const onCCPComplete = ({ name, lat, lng, address }) => {
+    if (mode === MapModes.NewCCP) {
+      addCCP({
+        variables: {
+          name,
+          eventId,
+          createdBy: 1,
+        },
+      }).then((addCCPData) => {
+        addPin({
+          variables: {
+            label: name,
+            eventId,
+            latitude: lat,
+            longitude: lng,
+            address,
+            pinType: PinType.CCP,
+            ccpId: addCCPData.data.addCollectionPoint.id,
+          },
+        });
+      });
+    } else if (mode === MapModes.EditCCP) {
+      const ccpPins = pins.filter(
+        (pin) => pin.ccpId && pin.pinType === PinType.CCP
+      );
+      const ccpPinId = ccpPins.filter((pin) => pin.ccpId.id === ccpId)[0].id;
+      editPin({
+        variables: {
+          id: ccpPinId,
+          label: name,
+          eventId,
+          latitude: lat,
+          longitude: lng,
+          address,
+        },
+      });
+      editCCP({
+        variables: {
+          id: ccpId,
+          name,
+          eventId,
+        },
+      });
+    }
+  };
+
   return (
     <>
-      <MenuAppBar pageTitle="Map" eventId={eventId} selectedMaps />
+      {mode === MapModes.Map && (
+        <MenuAppBar pageTitle="Map" eventId={eventId} selectedMaps />
+      )}
       <InfoWindow
         title={interestPinTitle}
         address={interestPinAddress}
         open={infoWindowOpen}
+        type={interestPinType}
         handleEditClick={onEditClicked}
         handleClose={onInfoWindowClose}
         handleDeleteClicked={onDeleteClick}
       />
-      <Sidebar
-        open={openSidebar}
-        onClose={onSidebarClose}
-        title={isEdit ? 'Edit a location pin' : 'Add a location pin'}
-        clickedAddress={tempMarkerClick ? tempMarkerAddress : undefined}
-        clickedLocation={tempMarkerClick ? tempMarkerLocation : undefined}
-        onSuggestionTempMarkerSet={onSuggestionTempMarkerSet}
-        setTempMarkerClick={() => {
-          setTempMarkerClick(false);
+      {sidebarRender && (
+        <Sidebar
+          open={openSidebar}
+          onClose={onSidebarClose}
+          mode={mode}
+          title={sidebarTitle}
+          clickedAddress={tempMarkerClick ? tempMarkerAddress : undefined}
```

Comment:
any values that are manually unset should be set to `null` rather than `undefined`
```suggestion
          clickedAddress={tempMarkerClick ? tempMarkerAddress : null}
```

Reply:
This results in a Typescript error. If a prop is not mandatory, then it has two types: one which is the type of the object when passed (string, number, custom type) and the other type is `undefined`. Using `null` doesn't conform to the typing, hence why I use `undefined`.